### PR TITLE
[swss] Enhance ARP Update to Call Sonic Cfggen Once

### PR DESCRIPTION
**Pull request by tahmed-dev**
This PR limited the number of calls to sonic-cfggen to one call
per iteration instead of current 3 calls per iteration. 

The PR also installs jq on host for future scripts if needed.

signed-off-by: Tamer Ahmed <tamer.ahmed@microsoft.com>

**- Why I did it**
Reduce number of calls into `sonic-cfggen`
Fix bug with INTERFACE keyword
**- How I did it**
Used template when calling `sonic-cfggen`

**- How to verify it**
run the script manually with -x option
```
root@str-s6000-acs-14:/# ./arp_update 
+ ARP_UPDATE_VARS_FILE=/usr/share/sonic/templates/arp_update_vars.j2
+ /bin/true
++ sonic-cfggen -d -t /usr/share/sonic/templates/arp_update_vars.j2
+ ARP_UPDATE_VARS='{
    "interface": "",
    "pc_interface" : "PortChannel0001 PortChannel0002 PortChannel0003 PortChannel0004 ",
    "vlan" : "Vlan1000"
}'
++ echo '{' '"interface":' '"",' '"pc_interface"' : '"PortChannel0001' PortChannel0002 PortChannel0003 PortChannel0004 '",' '"vlan"' : '"Vlan1000"' '}'
++ jq -r .interface
+ INTERFACE=
++ echo '{' '"interface":' '"",' '"pc_interface"' : '"PortChannel0001' PortChannel0002 PortChannel0003 PortChannel0004 '",' '"vlan"' : '"Vlan1000"' '}'
++ jq -r .pc_interface
+ PC_INTERFACE='PortChannel0001 PortChannel0002 PortChannel0003 PortChannel0004 '
+ ALL_INTERFACE=' PortChannel0001 PortChannel0002 PortChannel0003 PortChannel0004 '
+ for intf in $ALL_INTERFACE
+ ping6cmd='ping6 -I PortChannel0001 -n -q -i 0 -c 1 -W 0 ff02::1 >/dev/null'
++ ip link show PortChannel0001
++ grep 'state UP'
+ intf_up='5: PortChannel0001: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000'
+ [[ -n 5: PortChannel0001: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000 ]]
+ eval ping6 -I PortChannel0001 -n -q -i 0 -c 1 -W 0 ff02::1 '>/dev/null'
++ ping6 -I PortChannel0001 -n -q -i 0 -c 1 -W 0 ff02::1
ping6: Warning: source address might be selected on device other than PortChannel0001.
+ for intf in $ALL_INTERFACE
+ ping6cmd='ping6 -I PortChannel0002 -n -q -i 0 -c 1 -W 0 ff02::1 >/dev/null'
++ ip link show PortChannel0002
++ grep 'state UP'
+ intf_up='6: PortChannel0002: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000'
+ [[ -n 6: PortChannel0002: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000 ]]
+ eval ping6 -I PortChannel0002 -n -q -i 0 -c 1 -W 0 ff02::1 '>/dev/null'
++ ping6 -I PortChannel0002 -n -q -i 0 -c 1 -W 0 ff02::1
ping6: Warning: source address might be selected on device other than PortChannel0002.
+ for intf in $ALL_INTERFACE
+ ping6cmd='ping6 -I PortChannel0003 -n -q -i 0 -c 1 -W 0 ff02::1 >/dev/null'
++ ip link show PortChannel0003
++ grep 'state UP'
+ intf_up='7: PortChannel0003: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000'
+ [[ -n 7: PortChannel0003: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000 ]]
+ eval ping6 -I PortChannel0003 -n -q -i 0 -c 1 -W 0 ff02::1 '>/dev/null'
++ ping6 -I PortChannel0003 -n -q -i 0 -c 1 -W 0 ff02::1
ping6: Warning: source address might be selected on device other than PortChannel0003.
+ for intf in $ALL_INTERFACE
+ ping6cmd='ping6 -I PortChannel0004 -n -q -i 0 -c 1 -W 0 ff02::1 >/dev/null'
++ ip link show PortChannel0004
++ grep 'state UP'
+ intf_up='8: PortChannel0004: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000'
+ [[ -n 8: PortChannel0004: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9100 qdisc noqueue state UP mode DEFAULT group default qlen 1000 ]]
+ eval ping6 -I PortChannel0004 -n -q -i 0 -c 1 -W 0 ff02::1 '>/dev/null'
++ ping6 -I PortChannel0004 -n -q -i 0 -c 1 -W 0 ff02::1
ping6: Warning: source address might be selected on device other than PortChannel0004.
++ echo '{' '"interface":' '"",' '"pc_interface"' : '"PortChannel0001' PortChannel0002 PortChannel0003 PortChannel0004 '",' '"vlan"' : '"Vlan1000"' '}'
++ jq -r .vlan
+ VLAN=Vlan1000
+ for vlan in $VLAN
+ arpingcmd='sed -e '\''s/ / -i /'\'' -e '\''s/^/arping -q -w 0 -c 1 /'\'' -e '\''s/$/;/'\'''
+ ipcmd='ip -4 neigh show | grep Vlan1000 | cut -d '\'' '\'' -f 1,3 | sed -e '\''s/ / -i /'\'' -e '\''s/^/arping -q -w 0 -c 1 /'\'' -e '\''s/$/;/'\'''
++ eval ip -4 neigh show '|' grep Vlan1000 '|' cut -d \' \' -f 1,3 '|' sed -e ''\''s/' / -i '/'\''' -e ''\''s/^/arping' -q -w 0 -c 1 '/'\''' -e ''\''s/$/;/'\'''
+++ ip -4 neigh show
+++ grep Vlan1000
+++ sed -e 's/ / -i /' -e 's/^/arping -q -w 0 -c 1 /' -e 's/$/;/'
+++ cut -d ' ' -f 1,3
+ eval
+ ping6cmd='ping6 -I Vlan1000 -n -q -i 0 -c 1 -W 0 ff02::1 >/dev/null'
+ eval ping6 -I Vlan1000 -n -q -i 0 -c 1 -W 0 ff02::1 '>/dev/null'
++ ping6 -I Vlan1000 -n -q -i 0 -c 1 -W 0 ff02::1
ping6: Warning: source address might be selected on device other than Vlan1000.
+ ndisc6cmd='sed -e '\''s/^/ndisc6 -q -w 0 -1 /'\'' -e '\''s/$/;/'\'''
+ ip6cmd='ip -6 neigh show | grep -v fe80 | grep Vlan1000 | cut -d '\'' '\'' -f 1,3 | sed -e '\''s/^/ndisc6 -q -w 0 -1 /'\'' -e '\''s/$/;/'\'''
++ eval ip -6 neigh show '|' grep -v fe80 '|' grep Vlan1000 '|' cut -d \' \' -f 1,3 '|' sed -e ''\''s/^/ndisc6' -q -w 0 -1 '/'\''' -e ''\''s/$/;/'\'''
+++ ip -6 neigh show
+++ grep -v fe80
+++ grep Vlan1000
+++ cut -d ' ' -f 1,3
+++ sed -e 's/^/ndisc6 -q -w 0 -1 /' -e 's/$/;/'
+ eval
+ sleep 300
```

**- Which release branch to backport (provide reason below if selected)**
- [ ] 201811
- [x] 201911
- [x] 202006

### CODE REVIEW

**Comment by tahmed-dev** [comment]
retest mellanox please

**Comment by tahmed-dev** [comment]
retest vsimage please

**Comment by tahmed-dev** [comment]
retest mellanox please

**Comment by tahmed-dev** [comment]
retest vsimage please

**Review by pavel-shirshov — DISMISSED** [review]
LGTM

File: files/scripts/arp_update [hunk 1]
```diff
@@ -7,12 +7,13 @@
...
 
+ARP_UPDATE_VARS_FILE="/usr/share/sonic/templates/arp_update_vars.j2"
+
 while /bin/true; do
   # find L3 interfaces which are UP, send ipv6 multicast pings
-  echo "{% for (name, prefix) in INTERFACE|pfx_filter %} {{name}} {% endfor %}" > /tmp/intf_tmp.j2
-  INTERFACE=`sonic-cfggen -d -t /tmp/intf_tmp.j2`
-  echo "{% for (name, prefix) in PORTCHANNEL_INTERFACE|pfx_filter %} {{name}} {% endfor %}" > /tmp/pc_intf_tmp.j2
-  PC_INTERFACE=`sonic-cfggen -d -t /tmp/pc_intf_tmp.j2`
+  ARP_UPDATE_VARS=$(sonic-cfggen -d -t ${ARP_UPDATE_VARS_FILE})
```
Comment (prsunny) [inline files/scripts/arp_update:14]: `INTERFACE `is from  INTERFACE table and not PORT and `PC_INTERFACE `is from PORTCHANNEL_INTERFACE and not PORTCHANNEL. Seems there is a mismatch in the `arp_update_vars.j2`

Comment (tahmed-dev) [reply files/scripts/arp_update:14]: OK, my config_db on the dut did not have INTERFACE table. Are we looking here for interface that are not portchannels? If not, can you please let me know what the INTERFACE contents are? Also, for PORTCHANNEL_INTERFACE, it has repeated contents that appeared unnecessary to me.

Comment (prsunny) [reply files/scripts/arp_update:14]: The contents of INTERFACE would be same as PORTCHANNEL_INTERFACE. Both have IP addresses in key. Then "unnecessary" contents are filtered by this - `PORTCHANNEL_INTERFACE|pfx_filter`

Comment (tahmed-dev) [reply files/scripts/arp_update:14]: I don't see INTERFACE entry in the config_db.json. I tried a couple of devices. Where can I find it. The original script had interface generates empty list. For PORTCHANNEL_INTERFACE, there were duplicate entries in the pc_interface list. Before the ping6cmd is run, the interface is checked for `state UP`. There are differences, however is the new way of extracting info leading to the correct interfaces being extracted?

Comment (prsunny) [reply files/scripts/arp_update:14]: No, the new way of extracting doesn't give the correct interfaces

**Review by prsunny — APPROVED** [review]

**Comment by tahmed-dev** [comment]
retest baseimage please

**Pull request merged**